src/service-worker/ServiceWorker.ts

```diff
@@ -375,6 +378,19 @@ export class ServiceWorker {
     return await OneSignalApiBase.put(`notifications/${notification.id}/report_received`, postData);
   }
 
+  /**
+   * Confirmed Delivery isn't supported on Safari since they are very strict about the amount
+   * of time you have to finish the onpush event. Spending to much time in the onpush event
+   * will cause the push endpoint to become revoked!, causing the device to stop receiving pushes!
+   *
+   * iPadOS 16.4 it was observed to be only about 10 secounds.
+   * macOS 13.3 didn't seem to have this restriction when testing up to a 25 secound delay, however
+   * to be safe we are disabling it for all Safari browsers.
```

Comment:
This would essentially kill RR for all existing Safari subscriptions. I wonder if it's a better idea to only add the restriction to Safari on mobile.

You could use navigator.platform to detect this.

Reply:
>This would essentially kill RR for all existing Safari subscriptions.

We have never had a release where we enabled macOS Safari VAPID push (device_type = 17) yet. For Safari Legacy push (device_type = 7) this code path would never happen, since pushes are not displayed by the service worker, but directly by Safari itself.

> I wonder if it's a better idea to only add the restriction to Safari on mobile.

I thought of that, but since Apple didn't state any different limits between Safari mobile vs desktop there is a chance Apple could start enforcing this for desktop later. It is much safer to disable it for all Safari types, there is some details released by Apple stating the differences, if it was even intended.

> You could use navigator.platform to detect this.

I did try this, I put my notes in the "Other Options Considered" section of this PR.